net-test/bin/start.sh

```diff
@@ -0,0 +1,363 @@
+   logln "PID $FAUCET_PID"
+
+   echo "$FAUCET_PID"
+   return 0
+}
+
+start_stacks_master_node() {
+   logln "Setting up Stacks master node..."
+   test -f "$STACKS_MASTER_CONF".in || exit_error "No such file or directory: $STACKS_MASTER_CONF.in"
+
+   log "  Generating Stacks master config file..."
+   sed -r \
+      -e "s!@@STACKS_CHAINSTATE_DIR@@!$STACKS_CHAINSTATE_DIR!g" \
+      -e "s!@@STACKS_PUBLIC_IP@@!$STACKS_MASTER_PUBLIC_IP!g" \
+      "$STACKS_MASTER_CONF".in \
+      > "$STACKS_MASTER_CONF"
+   logln "ok"
+
+   log "Starting Stacks master node..."
+   BLOCKSTACK_DEBUG=1 RUST_BACKTRACE=full stacks-node start --config="$STACKS_MASTER_CONF" >"$STACKS_MASTER_LOGFILE" 2>&1 &
+   local STACKS_PID=$!
+   logln "PID $STACKS_PID"
+
+   echo "$STACKS_PID"
+   return 0
+}
+
+start_stacks_miner_node() {
+   logln "Setting up Stacks miner node..."
+   test -f "$STACKS_MINER_CONF".in || exit_error "No such file or directory: $STACKS_MINER_CONF.in"
+
+   log "  Generating Stacks miner config file..."
+   local STACKS_MASTER_IP="$1"
+   local BITCOIN_IP="$2"
+   local FAUCET_URL="$3"
+   local STACKS_MINER_SEED="$(blockstack-cli generate-sk | jq -r '.secretKey')"
+
+   sed -r \
+      -e "s/@@STACKS_MASTER_IP@@/$STACKS_MASTER_IP/g" \
+      -e "s/@@BITCOIN_IP@@/$BITCOIN_IP/g" \
+      -e "s/@@STACKS_MINER_SEED@@/$STACKS_MINER_SEED/g" \
+      -e "s!@@STACKS_CHAINSTATE_DIR@@!$STACKS_CHAINSTATE_DIR!g" \
+      "$STACKS_MINER_CONF".in \
+      > "$STACKS_MINER_CONF"
+   logln "ok"
+
+   local BTCADDR="$(blockstack-cli --testnet addresses "$STACKS_MINER_SEED" | jq -r '.BTC')"
+   local TIMEOUT=2
+   local TXID=""
+   local CONFIRMATIONS=0
+
+   log "  Fetching BTC from the faucet for this miner ($BTCADDR)..."
+   for i in $(seq 1 10); do
+       TXID="$(curl -sf -X POST -d $BTCADDR$'\n' -H "content-type: text/plain" "$FAUCET_URL"/fund)"
+       local RC=$?
+       if [ $RC -ne 0 ]; then 
+          # curl failed, or we didn't get 200.  Try again
+          logln "FAILED!  Try again in $TIMEOUT seconds..."
+          sleep $TIMEOUT
+
+          TIMEOUT=$((TIMEOUT + $RANDOM % TIMEOUT))
+          continue
+       fi
+       break
+   done
+   logln "txid $TXID"
+
+   log "  Waiting for BTC to get confirmed..."
+   for i in $(seq 1 5); do
+       sleep 60
+       local CONFIRMATIONS="$(curl -sf "$FAUCET_URL"/confirmations/"$TXID")"
+       local RC=$?
+       if [ $RC -ne 0 ]; then 
+          logln "FAILED!  Try again (attempt $i)..."
+          continue
+       fi
+       if [ $CONFIRMATIONS -gt 0 ]; then 
+          break
+       fi
+   done
+
+   if [ $CONFIRMATIONS -eq 0 ]; then 
+      logln "FAILED"
+      return 1
+   fi
+   logln "ok"
+   
+   log "Starting Stacks miner node..."
+   BLOCKSTACK_DEBUG=1 RUST_BACKTRACE=full stacks-node start --config="$STACKS_MINER_CONF" >"$STACKS_MINER_LOGFILE" 2>&1 &
+   local STACKS_PID=$!
+   logln "PID $STACKS_PID"
+
+   echo "$STACKS_PID"
+   return 0
+}
+
+start_stacks_follower_node() {
```

Comment:
Note to reviewers: I haven't tested this method yet.

Reply:
seemed to startup correctly for me